libs/net/template/ReportEngine.cs

```diff
@@ -352,6 +361,92 @@ public async Task<string> GenerateReportBodyAsync(
         var template = this.ReportEngineContent.GetOrAddTemplateInMemory(key, report.Template.Body)
             ?? throw new InvalidOperationException("Template does not exist");
 
+        // For each section that is JSON from 3rd party, fetch the JSON and save it.
+        await report.Sections.Where(section => section.SectionType == Entities.ReportSectionType.Data && section.IsEnabled).ForEachAsync(async section =>
+        {
+            var sectionData = sectionContent[section.Name];
+            var settings = section.Settings;
+            var url = !String.IsNullOrWhiteSpace(settings.Url) ? new Uri(settings.Url) : null;
+            if (url != null)
+            {
+                var response = await this.HttpClient.GetAsync(url);
```

Comment:
We make requests for external data and put it into the report when the report is generated.